python/pyspark/ml/tests.py

```diff
@@ -71,6 +71,34 @@
 ser = PickleSerializer()
 
 
+def generate_multinomial_logistic_input(
+        weights, x_mean, x_variance, add_intercept, n_points, seed=None):
+    """Creates multinomial logistic dataset"""
+
+    if seed:
+        np.random.seed(seed)
+    n_features = x_mean.shape[0]
+
+    x = np.random.randn(n_points, n_features)
+    x = x * np.sqrt(x_variance) + x_mean
+
+    if add_intercept:
+        x = np.hstack([x, np.ones((n_points, 1))])
+
+    # Compute margins
+    margins = np.hstack([np.zeros((n_points, 1)), x.dot(weights.T)])
+    # Shift to avoid overflow and compute probs
+    probs = np.exp(np.subtract(margins, margins.max(axis=1).reshape(n_points, -1)))
+    # Compute cumulative prob
+    cum_probs = np.cumsum(probs / probs.sum(axis=1).reshape(n_points, -1), axis=1)
+    # Asign class
```

Comment:
"Assign class", though IMO you could also just do away with the comments in this section.